### ISSUE TITLE
[Feature]: Moderation API

### ISSUE THREAD

**Issue by kishanios123** [zilliztech/GPTCache#366]
### Is your feature request related to a problem? Please describe.

I am not able to use Moderation API.

### Describe the solution you'd like.

Add moderation api openai and also another possible solution, and add caching mechanism too.

**Comment by SimFG** [comment]
good idea, recently i think about the Moderation path. If you have additional background knowledge on this topic, we welcome your contributions.

**Comment by xiaofan-luan** [comment]
what is  Moderation API? any background

**Comment by kishanios123** [comment]
> what is Moderation API? any background

https://platform.openai.com/docs/guides/moderation/overview

**Comment by kishanios123** [comment]
> good idea, recently i think about the Moderation path. If you have additional background knowledge on this topic, we welcome your contributions.

Thank you for considering my request, but I am not able to contribute because I am a beginner in Python. I work on a different platform.
